src/rollup/plugins/server-assets.ts

```diff
@@ -71,7 +71,7 @@ for (const asset of serverAssets) {
 }
 
 function normalizeKey (key) {
-  return key.replace(/[/\\]/g, ':').replace(/^:|:$/g, '')
+  return key.replace(/[/\\]/g, ':').replace(/:+/g, ':').replace(/^:|:$/g, '')
```

Comment:
I don't know if this would be faster:

```suggestion
  return key.replace(/[/\\]/g, ':').replace(/:{2,}/g, ':').replace(/^:|:$/g, '')
```